src/app/rpmostree-composeutil.c

```diff
@@ -249,10 +249,16 @@ rpmostree_composeutil_get_treespec (RpmOstreeContext  *ctx,
     return FALSE;
   if (!treespec_bind_array (treedata, treespec, "exclude-packages", NULL, FALSE, error))
     return FALSE;
-  if (!treespec_bind_array (treedata, treespec, "repos", NULL, TRUE, error))
+  if (!treespec_bind_array (treedata, treespec, "repos", NULL, FALSE, error))
     return FALSE;
   if (!treespec_bind_array (treedata, treespec, "lockfile-repos", NULL, FALSE, error))
     return FALSE;
+
+  /* at least one of `repos` and `lockfile-repos` should be defined */
+  if (!json_object_has_member (treedata, "repos") &&
+      !json_object_has_member (treedata, "lockfile-repos"))
+    return glnx_null_throw (error, "Treefile has neither \"repos\" nor \"lockfile-repos\" members");
```

Comment:
`glnx_throw`

Reply:
This function actually returns a pointer, not a boolean. I added a commit to convert all the `return FALSE` in here to `return NULL`.